Examples/Makefile.in

```diff
@@ -904,13 +904,12 @@ NOLINK ?= false
 OCAMLPP= -pp "camlp4o ./swigp4.cmo"
 OCAMLP4WHERE=`$(COMPILETOOL) @CAMLP4@ -where`
 OCAMLCORE=\
-	rm -rf swig.mli swig.ml swigp4.ml && \
-	$(SWIG) -ocaml -co swig.mli 2>/dev/null &&  \
-	$(SWIG) -ocaml -co swig.ml 2>/dev/null &&  \
-	$(SWIG) -ocaml -co swigp4.ml 2>/dev/null &&  \
-	$(OCC) -c swig.mli &&  \
-	$(OCC) -c swig.ml &&  \
-	$(OCC) -I $(OCAMLP4WHERE) -pp "camlp4o pa_extend.cmo q_MLast.cmo" -c swigp4.ml
+	(test -f swig.mli   || $(SWIG) -ocaml -co swig.mli 2>/dev/null) &&  \
+	(test -f swig.ml    || $(SWIG) -ocaml -co swig.ml 2>/dev/null) &&  \
+	(test -f swigp4.ml  || $(SWIG) -ocaml -co swigp4.ml 2>/dev/null) &&  \
+	(test -f swig.cmi   || $(OCC) -c swig.mli) &&  \
+	(test -f swig.cmo   || $(OCC) -c swig.ml) &&  \
+	(test -f swigp4.cmi || $(OCC) -I $(OCAMLP4WHERE) -pp "camlp4o pa_extend.cmo q_MLast.cmo" -c swigp4.ml)
```

Comment:
Isn't this `test` redundant now for the test-suite?
For the examples it should always be run otherwise it may be out of date.

Reply:
> Isn't this test redundant now for the test-suite?

No, it's not redundant: OCAMLCORE is always run before each test case, so the `test`s are needed to prevent unnecessary operations (and race conditions).

> For the examples it should always be run otherwise it may be out of date.

It seems that would only occur if someone was tinkering with the swig library files.
I had felt that one should run `make clean-ocaml-examples`  or `make clean` in the individual example dirs to remove the old/outdated files.
So, I'm not sure if a change should be made here to the proposed code.